gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/common/impl/PathUtils.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.config.common.impl;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import org.apache.hadoop.fs.Path;
+
+
+/*
+ * Used to resolve the hadoop 1 -> 2 issue
+ */
+public class PathUtils {
+
+  public static Path getParent(URI uri) {
+    if (uri == null)
+      return null;
+
+    String pStr = uri.getPath();
+    if (pStr.length() == 0)
+      return null;
+
+    Path p = (new Path(uri.getPath())).getParent();
+    return p;
+  }
+
+  public static URI getParentURI(URI uri) {
+    String pStr = uri.getPath();
+    if (pStr.length() == 0)
+      return null;
+
+    Path p = (new Path(uri.getPath())).getParent();
+    try {
+      return new URI(uri.getScheme(), uri.getAuthority(), p.toString(), uri.getQuery(), uri.getFragment());
+    } catch (URISyntaxException e) {
+      // Should not come here
+      e.printStackTrace();
+    }
+    return null;
+  }
+
+  /**
+   * @param parent - parent URI
+   * @param child - descendant URI
+   * @return true if o2 is the descendant of o1 
+   */
+  public static boolean checkDescendant(URI parent, URI child) {
+    if (parent == null && child == null) {
+      return true;
+    }
+
+    if (parent == null || child == null) {
+      return false;
+    }
+
+    boolean result = false;
+    String scheme1 = parent.getScheme();
+    String scheme2 = child.getScheme();
+    result = stringEquals(scheme1, scheme2);
+    if (!result)
+      return result;
+
+    String auth1 = parent.getAuthority();
+    String auth2 = child.getAuthority();
+    result = stringEquals(auth1, auth2);
+    if (!result)
+      return result;
+
+    String path1 = parent.getPath();
+    String path2 = child.getPath();
+
+    // path2 starts with path1
+    if (path2.equals(path1) || path2.indexOf(path1 + "/") == 0) {
```

Comment:
if path2 is /user/mitu/foobar and path1 is /user/mitu/foo, then this will not work

Added more checks into this class